salt/metalk8s/addons/dex/deployed/service-configuration.sls

```diff
@@ -0,0 +1,66 @@
+include:
+  - .namespace
+
+{%- set dex_config = salt.metalk8s_kubernetes.get_object(
+        kind='ConfigMap',
+        apiVersion='v1',
+        namespace='metalk8s-auth',
+        name='metalk8s-dex-config'
+  )
+%}
+
+{%- if dex_config is none %}
+
+Create dex-config ConfigMap:
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: metalk8s-dex-config
+          namespace: metalk8s-auth
+        data:
+          config.yaml: |-
+            apiVersion: addons.metalk8s.scality.com
+            kind: DexConfig
+            spec:
+              deployment:
+                replicas: 2
+              localuserstore:
+                enabled: true
+                email: "admin@metalk8s.invalid"
+                hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
+              connectors:
+                enabled: false
+                type: ldap
```

Comment:
I don't like having this in the 'default' config. The 'default' config should represent the config that's running and not include 'stuff' with placeholders that's not used at all. That ought to be found in documentation.

Reply:
Agreed!